src/components/ComunityPicker/index.js

```diff
@@ -0,0 +1,157 @@
+
+const ComunityPicker = ({visible, toggle, setComunity, update}) => {
+  const [itens, setItens] = useState([]);
+  const [auxItens, setAuxItens] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+  useEffect(() => {
+    async function getListCommunity() {
+      await sleep(1000);
+      const response = await api
+        .get('community/listCommunities')
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          if (
+            error.response.data.message === 'Não há Comunidades cadastradas'
+          ) {
+            setItens([]);
+            setAuxItens([]);
+            setLoading(false);
+          } else {
+            if (visible) {
+              Alert.alert(
+                'Atenção!',
+                'Erro ao buscar dados. Tente novamente mais tarde!',
+              );
+            }
+            setLoading(true);
+            if (visible) {
+              toggle();
+            }
+          }
+        });
+      if (response) {
+        await sleep(1000);
+        const newItens = [...response.data];
+
+        newItens.sort((a, b) => {
+          return a.name > b.name ? 1 : -1;
+        });
+        setItens(newItens);
+        setAuxItens(newItens);
+        setLoading(false);
+      }
+    }
+    getListCommunity();
+  }, [update]);
+
+  const renderItem = (item) =>
+    loading ? (
+      <View style={{flex: 1}}>
+        <ActivityIndicator size={30} color="#ccc" />
+      </View>
+    ) : (
+      <UserItem
+        onPress={() => {
+          setComunity(item);
+          setAuxItens([...itens]);
+          setLoading(true);
+          toggle();
+        }}>
+        <Icon size={normalize(25)} name="user-circle" color="#a3a3a3" />
+        <ItemText>{item.name}</ItemText>
+      </UserItem>
+    );
+
+  const findResults = (text) => {
+    setAuxItens(
+      itens.filter(
+        (item) => item.name.toUpperCase().indexOf(text.toUpperCase()) >= 0,
+      ),
+    );
+  };
+
+  const onCloseModal = () => {
+    setLoading(true);
+    toggle();
+    setItens([]);
+    setAuxItens([]);
+  };
+
+  return (
+    <Modal
+      isVisible={visible}
+      style={{justifyContent: 'flex-end', alignItems: 'center', margin: 0}}
+      onPress>
+      <ModalContainer>
+        {!loading && itens.length === 0 ? (
+          <MessageText>
+            Não existem usuários que não estão em comunidades!
+          </MessageText>
+        ) : (
+          <>
+            <SearchBox>
+              <Input
+                label="Pesquisar por comunidades"
+                onChangeText={(text) => findResults(text)}
+                autoCapitalize="words"
+              />
+            </SearchBox>
+            <FlatListView>
+              <FlatList
+                keyboardShouldPersistTaps="handled"
+                mb={3}
+                verticalScroll
+                keyExtractor={(item) => item.email}
```

Comment:
O atributo `keyExtractor` da FlatList serve pra coletar algum dado único do item que está sendo renderizado, o objetivo disso é trazer uma performance melhor pro código, tem mais informações na [documentação do React Native ](https://reactnative.dev/docs/flatlist#keyextractor). Nesse caso vc ta renderizando uma comunidade, e a comunidade não tem o atributo `email`. Nesse caso todos os items vão ficar com a key `undefined`, e se tiver mais de uma comunidade provavelmente isso vai disparar um warning. Sugiro usar outro atributo aqui, como id 
```suggestion
                keyExtractor={(item) => item.id}
```